notesapi/v1/tests/test_views.py

```diff
@@ -387,9 +388,14 @@ def test_read_all_no_annotations(self):
         headers["course_id"] = "a/b/c"
         response = self.client.get(reverse('api:v1:annotations'), headers)
         self.assertEqual(response.status_code, status.HTTP_200_OK)
-        self.assertDictContainsSubset(
-            {'total': 0, 'rows': []}, response.data, "no annotation should be returned in response"
-        )
+        if sys.version_info[1] == 12:  # python 3.12
+            self.assertEqual(
```

Comment:
AssertEqual has always been present, why not just use that in both versions and not have a conditional?

Reply:
With python 3.8 we cannot use `|` operator with dicts
`E       TypeError: unsupported operand type(s) for |: 'dict' and 'dict'`

It was added in python 3.9 https://peps.python.org/pep-0584/
Error https://github.com/openedx/edx-notes-api/actions/runs/8018877609/job/21905671987#step:6:58

Reply:
Also there's no setup.py file in this repo.

Reply:
gotcha, thanks for the  context.